src/backend/distributed/transaction/worker_transaction.c

```diff
@@ -185,14 +196,19 @@ SendBareCommandListToWorkers(TargetWorkerSet targetWorkerSet, List *commandList)
  * always used and no transaction block is used (hence "bare").
  */
 int
-SendBareOptionalCommandListToWorkersAsUser(TargetWorkerSet targetWorkerSet,
-										   List *commandList, const char *user)
+SendBareOptionalCommandListToAllWorkersAsUser(List *commandList, const char *user)
 {
+	TargetWorkerSet targetWorkerSet = WORKERS_WITH_METADATA;
 	List *workerNodeList = TargetWorkerSetNodeList(targetWorkerSet, ShareLock);
 	ListCell *workerNodeCell = NULL;
 	ListCell *commandCell = NULL;
 	int maxError = RESPONSE_OKAY;
 
+	if (targetWorkerSet == WORKERS_WITH_METADATA)
+	{
+		ErrorIfAnyMetadataNodeOutOfSync(workerNodeList);
```

Comment:
Not sure this correct, given that this `if` statement is always true, and SendBareOptionalCommandListToAllWorkersAsUser is used for ALTER ENUM, which also applies to shards.

Reply:
I'm still thinking about this, there is no caller of `SendBareOptionalCommandListInternal` with metadata workers. I don't want any future caller to skip this.

In that sense, I'm planning to have an `SendBareOptionalCommandListToMetadataWorkersAsUser ()` and do not have any callers for that function. Or, would it be more confusing than not having `ErrorIfAnyMetadataNodeOutOfSync ` check in `SendBareOptionalCommandList*()` API?

Reply:
Maybe just remove the TargetWorkerSet argument?